proto/strings.proto

```diff
@@ -3,19 +3,39 @@ syntax = "proto3";
 package strings;
 
 service StringsExtractor {
-   rpc ExtractStrings (StringsRequest) returns (StringsReply) {}
+	rpc ExtractStrings (StringsRequest) returns (StringsReply) {}
 }
 
 message StringsRequest {
-   // ID of file.
-   string id = 1;
+	// Binary ID.
+	string bin_id = 1;
 }
 
 message StringInfo {
+	// File offset to start of string.
 	uint64 location = 1;
+	// String in UTF-8 encoding.
 	string raw_string = 2;
+	// Size of string in bytes.
+	uint64 size = 3;
```

Comment:
Are clients supposed to filter strings based on `raw_string.length()`? 

Also, refresh my memory, UTF-8 is capable of presenting any code point also present in UTF-16 and UTF-32, _right_? So there is a lossless conversion to UTF-8 from all other major formats?

Reply:
`raw_string.length()` returns the length in number of characters (Unicode code points) of the string.

The `size` field returns the size in number of bytes. E.g. UTF-16 takes between 2 and 4 bytes per Unicode code point. Two in the normal case and 4 for surrogate pairs.

How we do with `min_length` is yet to be decided. For now, the strings service simple returns all strings of all encodings that have length at least e.g. 4 number of Unicode code points in their string. It would be possible to push this logic to the client too, using `raw_string.length()`